### PR TITLE
Hardcode xcodebuild destination iOS simulator OS to 16.4.

### PR DESCRIPTION
**Pull request by edgchen1**
### Description

Hardcode xcodebuild destination iOS simulator OS to 16.4.

### Motivation and Context

Fix iOS CI build.

### CODE REVIEW

**Comment by skottmckay** [comment]
Possibly these places need updating as well:

https://github.com/microsoft/onnxruntime/blob/b8bbc898c6436076f2852a461cd273ecf2bca805/tools/ci_build/github/azure-pipelines/templates/react-native-ci.yml#L236

https://github.com/microsoft/onnxruntime/blob/b8bbc898c6436076f2852a461cd273ecf2bca805/tools/ci_build/github/azure-pipelines/mac-ios-packaging-pipeline.yml#L115

**Review by skottmckay — APPROVED** [review]

**Comment by edgchen1** [comment]
thanks for pointing these out.

> https://github.com/microsoft/onnxruntime/blob/b8bbc898c6436076f2852a461cd273ecf2bca805/tools/ci_build/github/azure-pipelines/templates/react-native-ci.yml#L236

this one is part of a job that uses the macOS-12 image so it should be fine.

> https://github.com/microsoft/onnxruntime/blob/b8bbc898c6436076f2852a461cd273ecf2bca805/tools/ci_build/github/azure-pipelines/mac-ios-packaging-pipeline.yml#L115

will update this one.

File: tools/ci_build/github/azure-pipelines/mac-ios-packaging-pipeline.yml [hunk 1]
```diff
@@ -112,6 +112,7 @@ stages:
         set -e -x
         cp "$(Pipeline.Workspace)/ios_packaging_artifacts_full/pod-archive-onnxruntime-c-$(ortPodVersion).zip" swift/
         export ORT_IOS_POD_LOCAL_PATH="swift/pod-archive-onnxruntime-c-$(ortPodVersion).zip"
-        xcodebuild test -scheme onnxruntime -destination 'platform=iOS Simulator,name=iPhone 14'
+        # hardcode iOS 16.4 for now. latest macOS-13 image defaults to iOS 17 (beta).
+        xcodebuild test -scheme onnxruntime -destination 'platform=iOS Simulator,OS=16.4,name=iPhone 14'
```
Comment (YUNQIUGUO) [inline tools/ci_build/github/azure-pipelines/mac-ios-packaging-pipeline.yml:116]: SPM repo is using macos-13 image agent too. assume I'll have to hardcode that to 16.4 as well?

**Review by YUNQIUGUO — APPROVED** [review]

File: tools/ci_build/github/azure-pipelines/mac-ios-packaging-pipeline.yml (hunk 1 above)
Comment (edgchen1) [reply tools/ci_build/github/azure-pipelines/mac-ios-packaging-pipeline.yml:116]: perhaps - I think this test didn't actually fail, but we don't want to be using a beta iOS version. that said, I'm working on a script to query the simulator info from xcode, so hopefully the hardcoded values can be replaced soon.

Comment (YUNQIUGUO) [reply tools/ci_build/github/azure-pipelines/mac-ios-packaging-pipeline.yml:116]: Got it. thanks!

**Review by skottmckay — APPROVED** [review]

**Pull request merged**